diesel/src/mysql/types/primitives.rs

```diff
@@ -0,0 +1,128 @@
+use std::error::Error;
+use std::str::{self, FromStr};
+
+use crate::deserialize::{self, FromSql};
+use crate::mysql::{Mysql, MysqlValue};
+use crate::sql_types::{BigInt, Binary, Double, Float, Integer, SmallInt, Text};
+
+fn decimal_to_integer<T>(bytes: &[u8]) -> deserialize::Result<T>
```

Comment:
I think we should be offloading part of this function to `BigDecimal::from_bytes` which should check all the possible errors for us. And then we should retrieve the `T` from it.

Reply:
`bigdecimal` is currently an optional dependency. In my opinion that should stay that way if possible.